s3-build-and-deploy.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# text output formatting vars
+YAY=$(tput setaf 47) # for good events
+INFO=$(tput setaf 39) # for important events
+CLI=$(tput setaf 214) # for code and commands
+BARF=$(tput setaf 1) # for bad events
+PLAIN=$(tput sgr0) # for everything else
+
+function info()
+{
+    echo "${INFO}$1${PLAIN}"
+}
+
+function cli()
+{
+    echo "${CLI}$1${PLAIN}"
+}
+
+function yay()
+{
+    echo "${YAY}$1${PLAIN}"
+}
+
+function barf()
+{
+    echo "${BARF}$1${PLAIN}"
+}
+
+function cleanup()
+{
+  if [ -n "${UNIQUE_WHEEL+x}" ]; then
+    rm -f "$UNIQUE_WHEEL"
+  fi
+}
+
+# fail fast on any error as a best practice, for more information see:
+# https://buildkite.com/docs/pipelines/writing-build-scripts#configuring-bash
+set -euo pipefail
+
+# ensure that we always cleanup if the script is interrupted
+trap cleanup EXIT SIGINT SIGTERM
+
+S3_PACKAGE_BUCKET_NAME="deltacat-packages-$USER"
+S3_PACKAGE_BUCKET_URL="s3://$S3_PACKAGE_BUCKET_NAME"
+CURRENT_UNIX_TIME=$(date +"%s")
+
+info "user=$USER"
+info "current unix time=$CURRENT_UNIX_TIME"
+info "working directory=$(pwd)"
+info "creating s3 bucket @ $S3_PACKAGE_BUCKET_NAME"
+aws s3api create-bucket --bucket "$S3_PACKAGE_BUCKET_NAME"
+# cleanup old deltacat build artifacts and build a new deltacat wheel
+info "rebuilding deltacat wheels"
+make rebuild
+# copy the new deltacat wheel to a unique wheel name w/ timestamp build tag
+pushd dist
+for file in deltacat-*.whl
+do
+    if [ -z ${UNIQUE_WHEEL+x} ]; then
+      UNIQUE_WHEEL=${file//-py3-none-any/-$CURRENT_UNIX_TIME-py3-none-any}
+      info "copying $file to $UNIQUE_WHEEL"
+      cp "$file" "$UNIQUE_WHEEL"
+    else
+      barf "ERROR: multiple deltacat wheels found: ${ls dist/deltacat-*.whl}"
+      exit 1
+    fi
+done
+# upload the new deltacat wheel to s3 and create a presigned URL for it
+info "uploading $UNIQUE_WHEEL to $S3_PACKAGE_BUCKET_URL"
+aws s3 cp "$UNIQUE_WHEEL" "$S3_PACKAGE_BUCKET_URL"
```

Comment:
This script assumes that user's environment has aws cli installed but doesn't look like Makefile has it. Can you confirm?

Reply:
The AWS CLI needs to be installed as a separate development environment setup step. The process here will be documented (i.e. pulled up from the `glue_runner.py` script into `README-development.md`) in a follow-up PR to update developer documentation.